server/protocols/webadmin/webadmin-rabbitmq/src/main/java/org/apache/james/webadmin/routes/RabbitMQRoutes.java

```diff
@@ -0,0 +1,165 @@
+/****************************************************************
```

Comment:
I propose the following endpoint:

```
curl -XPATCH /mailQueues/name/mails?task=rePublishOnRabbitMQ&notDequeuedAfter=2hours
```

 - We act on the mail entities
 - PATCH as we alter their state
 - rePublishOnRabbitMQ better describes the operation than "rebuild"
 - We need the notDequeuedAfter to be explicit.

Do we agree on the API?

We do something similar for the *flush* operation so it is consistent..

Reply:
Note that you need to define an extention mechanism for `curl -XPATCH /mailQueues/name/mails?...`

You can do that by:
 - Define a Set<TaskRegistration> that you inject.
 - Make the RabbitMQ webadmin project define the TaskRegistration, that you would then inject.
 - Wrap todays

Reply:
for me it is a POST as we submit a request and get the id of the newly created task in the response.

Reply:
Post if you want, I wont be able to argue much against it.